src/microbe_stage/MicrobeSpecies.cs

```diff
@@ -114,6 +114,13 @@ public override void UpdateInitialCompounds()
 
         foreach (var compoundBalance in compoundBalances)
         {
+            if (compoundBalance.Key == SimulationParameters.Instance.GetCompound("glucose") &&
```

Comment:
You should look up the compound just once at the start of the method to avoid this lookup by string key each iteration of the loop. Also the organelles count check and IsBacteria should be done before the loop. Something like `bool giveBonusGlucose = Organelles.Count <= 3 && IsBacteria`